src-django/api/xml_importer.py

```diff
@@ -0,0 +1,207 @@
+from django.db import transaction
+from xml.etree import ElementTree
+
+from api.models import Concept, Element, Page, Procedure, ShowIf
+
+import json
+import uuid
+
+
+class AttributeExtractor:
+    @staticmethod
+    def extract(element, attributes):
+        extracted_dict = {}
+        for attr_name, field_name in attributes.iteritems():
+            if attr_name in element.attrib:
+                extracted_dict[field_name] = element.attrib[attr_name]
+
+        return extracted_dict
+
+
+class ProcedureCreator:
+    ELEMENT_NAME = 'Procedure'
+    _ATTRIBUTES = {
+        'title': 'title',
+        'author': 'author',
+        'uuid': 'uuid',
+    }
+
+    @classmethod
+    def create_from(cls, owner, procedure_node):
+        fields_dict = AttributeExtractor.extract(procedure_node, cls._ATTRIBUTES)
+        fields_dict['owner'] = owner
+
+        if 'uuid' in fields_dict:
+            fields_dict['uuid'] = cls._parse_uuid(fields_dict['uuid'])
+
+        return Procedure(**fields_dict)
+
+    @classmethod
+    def _parse_uuid(cls, uuid_str):
+        return uuid.UUID(uuid_str).hex
+
+
+class PageCreator:
+    ELEMENT_NAME = 'Page'
+
+    @classmethod
+    def create_from(cls, procedure_id, display_index):
+        fields_dict = {
+            'display_index': display_index,
+            'procedure_id': procedure_id,
+        }
+        return Page(**fields_dict)
+
+
+class ElementCreator:
+    ELEMENT_NAME = 'Element'
+    _ATTRIBUTES = {
+        'type': 'element_type',
+        'concept': 'concept',
+        'question': 'question',
+        'answer': 'answer',
+        'required': 'required',
+        'image': 'image',
+        'audio': 'audio',
+        'action': 'action',
+        'mime_type': 'mime_type',
+        'choices': 'choices',
+    }
+
+    @classmethod
+    def create_from(cls, element_node, page_id, display_index):
+        fields_dict = AttributeExtractor.extract(element_node, cls._ATTRIBUTES)
+        fields_dict['page_id'] = page_id
+        fields_dict['display_index'] = display_index
```

Comment:
could be:
`
fields_dict.update({
    'page_id':page_id,
    'display_index':display_index,
})
`